### ISSUE TITLE
Aggregation Info's `=` operator calls Clear, before anything else, and segfaults.

### ISSUE THREAD

**Issue by kayabaNerve** [Chia-Network/bls-signatures#38]
Line 399 of AggregationInfo tries to delete a null element and causes a segfault.

My code likely won't be too useful as it's not in C++, but here's a link to my test that is demonstrating the issue: https://github.com/EmberCrypto/BLS/blob/adabf81f18343be9007abba61bf70868b47a247f/tests/SignatureTest.nim
And my library file that's wrapping the C++:
https://github.com/EmberCrypto/BLS/blob/adabf81f18343be9007abba61bf70868b47a247f/BLS/PrivateKey.nim#L94

@jonspock looked it over with me and we both believe it to not be a problem with Nim.

**Comment by jonspock** [comment]
I believe the following loops should only execute if the variables are not nulltpr/empty.
Using the operator= , there should only be a Clear() for pre-existing objects or re-code to avoid these loops (hope that is clear, if not, please ping me)

    for (auto &mapEntry : tree) {
        delete[] mapEntry.second;
        mapKeys.push_back(mapEntry.first);
    }
    for (const uint8_t* mapKey : mapKeys) {
        delete[] mapKey;
    }

**Comment by kayabaNerve** [comment]
Aren't these loops reducible to the below, anyways?
```
for (auto &mapEntry : tree) {
    delete[] mapEntry.first;
    delete[] mapEntry.second;
}
```

**Comment by mariano54** [comment]
Confirmed that this issue is fixed by #39?

**Comment by kayabaNerve** [comment]
Yep. I confirmed it before I made the PR.
